specs/FrameProcesssInfo.md

```diff
@@ -0,0 +1,503 @@
+
+* We propose extending `CoreWebView2FrameInfo` to include `FrameId`, 
+`FrameKind` and `ParentFrameInfo` properties. `FrameId` is the same 
+kind of ID as with `FrameID` in `CoreWebView2` and `CoreWebView2Frame`. 
+`ParentFrameInfo` supports to retrieve a frame's direct parent. This 
+also can be used to build the frame tree represented by `FrameInfo`s.
+
+# Examples
+C++
+```c++
+void AppendFrameInfo(
+    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo, std::wstringstream& result);
+wil::com_ptr<ICoreWebView2FrameInfo> GetAncestorFirstLevelFrameInfo(
+    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo);
+wil::com_ptr<ICoreWebView2FrameInfo> GetAncestorMainFrameInfo(
+    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo);
+std::wstring FrameKindToString(COREWEBVIEW2_FRAME_KIND kind);
+
+// Display renderer process info with details which includes the list of 
+// associated frame infos for the renderer process. Also shows the process 
+// info of other type of process.
+void ProcessComponent::ShowProcessFrameInfo()
+{
+    auto environment14 =
+        m_webViewEnvironment.try_query<ICoreWebView2Environment14>();
+    if (environment14)
+    {
+        //! [GetProcessInfosWithDetails]
+        CHECK_FAILURE(environment14->GetProcessInfosWithDetails(
+            Callback<ICoreWebView2GetProcessInfosWithDetailsCompletedHandler>(
+                [this](HRESULT error, ICoreWebView2ProcessInfoCollection* processCollection)
+                    -> HRESULT
+                {
+                    UINT32 processCount = 0;
+                    UINT32 rendererProcessCount = 0;
+                    CHECK_FAILURE(processCollection->get_Count(&processCount));
+                    std::wstringstream otherProcessInfos;
+                    std::wstringstream rendererProcessInfos;
+                    for (UINT32 i = 0; i < processCount; i++)
+                    {
+                        Microsoft::WRL::ComPtr<ICoreWebView2ProcessInfo> processInfo;
+                        CHECK_FAILURE(processCollection->GetValueAtIndex(i, &processInfo));
+                        COREWEBVIEW2_PROCESS_KIND kind;
+                        CHECK_FAILURE(processInfo->get_Kind(&kind));
+                        INT32 processId = 0;
+                        CHECK_FAILURE(processInfo->get_ProcessId(&processId));
+                        if (kind == COREWEBVIEW2_PROCESS_KIND_RENDERER)
+                        {
+                            //! [AssociatedFrameInfos]
+                            std::wstringstream rendererProcess;
+                            wil::com_ptr<ICoreWebView2ProcessInfo2> processInfo2;
+                            CHECK_FAILURE(
+                                processInfo->QueryInterface(IID_PPV_ARGS(&processInfo2)));
+                            wil::com_ptr<ICoreWebView2FrameInfoCollection> frameInfoCollection;
+                            CHECK_FAILURE(processInfo2->get_AssociatedFrameInfos(

```

Comment:
It will return an empty list,
>  `CoreWebView2ProcessInfo` objects obtained via `CoreWebView2Environment. GetProcessInfos` or for non-renderer processes will always have an empty `AssociatedFrameInfos`.

Reply:
This is a little confusing as there would be no way to guess that just from reading the api. 

Would it instead be possible to do something like:

```
runtimeclass CoreWebView2
{
    WebView2ProcessAndFrameInfo GetProcessAndFrameInfosAsync()
}

runtimeclass WebView2ProcessAndFrameInfo
{
    CoreWebView2ProcessInfo ProcessInfo {get; }
    IVectorView<CoreWebView2FrameInfo> FrameInfos { get; };
}

```
This way you would never end up with an object that included 'dummy' properties.

Reply:
Or `CoreWebView2ProcessInfo.GetAssociatedFrameInfos()`?

Reply:
This is another reason not to introduce GetProcessInfosWithDetails -  it fragments the behavior of GetProcessInfos.  Just add ICoreWebView2ProcessInfo2 with lazy evaluation of get_AssociatedFrameInfos.